sympy/matrices/matrices.py

```diff
@@ -1035,6 +1041,10 @@ def extract(self, rowsList, colsList):
         """
         cols = self.cols
         flat_list = self._mat
+        if len(rowsList) > 0 and isinstance(rowsList[0], bool):
```

Comment:
For all cases of these, should be `if rowsList and isinstance(rowsList[0], bool)`...